src/components/runtime-common.js

```diff
@@ -108,6 +108,18 @@ export const PeriodicCookieSetter = () => {
   return null
 }
 
+export const PeriodicAzureCookieSetter = ({ proxyUrl }) => {
+  const signal = useCancellation()
+  usePollingEffect(
+    withErrorIgnoring(async () => {
+      await Ajax(signal).Runtimes.setAzureCookie(proxyUrl)
```

Comment:
will this happen for all Azure runtimes the user has?

Reply:
the point where this is invoked is when you are launching an azure app, so a specific (running) runtime is present when this component is attached, since we need the specific proxyUrl associated with a runtime to call setCookie. If you have more than one runtime, there is logic to select a specific one [here](https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/runtime-utils.js#L282), which the cookie setter will use. 

See here for where the cookie setter is attached https://github.com/DataBiosphere/terra-ui/pull/3141/files#diff-44105412b7d7ce49a5ee9ac6b7202ea01d275dbd1054339f21abebdf163349e5R213